sklearnex/covariance/incremental_covariance.py

```diff
@@ -293,21 +294,22 @@ def _onedal_fit(self, X, queue=None):
         return self
 
     # expose sklearnex pairwise_distances if mahalanobis distance eventually supported
-    @wrap_output_data
     def mahalanobis(self, X):
         if sklearn_check_version("1.0"):
-            self._validate_data(X, reset=False, copy=self.copy)
-        else:
-            check_array(X, copy=self.copy)
+            self._check_feature_names(X, reset=False)
 
+        xp, _ = get_namespace(X)
```

Comment:
I think it make sense move out this the scopoe of the PR.
If this needed to be addressed related Array API, please mention it on #1861, #1813

Reply:
We discussed this elsewhere. will mark as resolved.